java/libs/src/main/java/org/physical_web/collection/PwsClient.java

```diff
@@ -61,8 +78,42 @@ public void setPwsEndpoint(String pwsEndpoint) {
     mPwsEndpoint = pwsEndpoint;
   }
 
-  private String constructPwsUrl(String path) {
-    return mPwsEndpoint + "/" + path;
+  /**
+   * Set the API key the PWS endpoint uses.
+   * @param pwsApiKey The new PWS endpoint.
+   */
+  public void setPwsApiKey(String pwsApiKey){
+      apiKey = pwsApiKey;
+  }
+
+  /**
+   * Set the API version the PWS endpoint uses.
+   * @param pwsApiVersion The new PWS endpoint.
+   */
+  public void setPwsApiVersion(int pwsApiVersion){
+    apiVersion = pwsApiVersion;
+  }
+
+  private String constructPwsResolveUrl() {
+    switch(apiVersion){
+      case 1:
+        return mPwsEndpoint + "/" + V1_RESOLVE_SCAN_PATH;
+      case 2:
+       return mPwsEndpoint + "/" + V2_RESOLVE_SCAN_PATH + "?key=" + apiKey;
+      default:
+        return "";
+    }
+  }
+
+  private String getPayloadApiJsonWrapper(){
```

Comment:
Style nit: you should have spaces after the params.  I think `./gradlew check` will point out style issues.